avocado/utils/data_structures.py

```diff
@@ -52,6 +53,63 @@ def geometric_mean(values):
     return math.exp(sum([math.log(number) for number in values]) / no_values)
 
 
+def compare_matrices(matrix1, matrix2, treshold=0.05):
+    """
+    Compare 2 matrices nxm and return a matrix nxm with comparison data
```

Comment:
Compare 2 matrices nxm and returns a matrix nxm with comparison data and stats. When the first columns match, they are considered as header and included in the results intact.